main.js

```diff
@@ -88,20 +86,49 @@ function showIdea() {
 }
 
 outputContainer.addEventListener('click', function(event) {
-    if (event.target.classList.contains('delete-new-idea-img')) { 
-        var idToDelete = event.target.closest('.new-idea').getAttribute('data-id');
-
-        var newIdeasArray = [];
-        for (var i = 0; i < currentIdeas.ideas.length; i++) {
-            if (currentIdeas.ideas[i].id.toString() !== idToDelete) {
-                newIdeasArray.push(currentIdeas.ideas[i]);
-            }
-        }
-        currentIdeas.ideas = newIdeasArray;
+  if (event.target.classList.contains('delete-new-idea-img')) { 
+    var idToDelete = event.target.closest('.new-idea').getAttribute('data-id');
+    var newIdeasArray = [];
 
-        var card = event.target.closest('.new-idea');
-        if (card) {
-            outputContainer.removeChild(card);
-        }
+    for (var i = 0; i < currentIdeas.ideas.length; i++) {
+      if (currentIdeas.ideas[i].id.toString() !== idToDelete) {
+        newIdeasArray.push(currentIdeas.ideas[i]);
+      }
+    }
+    currentIdeas.ideas = newIdeasArray;
+
+    var card = event.target.closest('.new-idea');
+    if (card) {
+      outputContainer.removeChild(card);
     }
-});
\ No newline at end of file
+  }
+});
+
+outputContainer.addEventListener('click', function(event) {
+  if (event.target.classList.contains("new-idea-fave-btn")) {
+    var faveIdeaId = event.target.closest('.new-idea').getAttribute('data-id');
+    for (var i = 0; i < currentIdeas.ideas.length; i++) {
+      if ((currentIdeas.ideas[i].id.toString() === faveIdeaId) && (currentIdeas.ideas[i].isFave === true)) {
+        currentIdeas.ideas[i].isFave = false;
+        for (var j = 0; j < faveIdeas.length; j++) {
+          if (faveIdeas[j].id.toString() === faveIdeaId) {
+            faveIdeas.splice(j, 1);
```

Comment:
We technically don't need the 1 as the 2nd argument there I think, but that's OK.

Reply:
I was basically mirroring what I saw in MDN, specifying that the count is only 1 for the splice, just to be more explicit.

Reply:
We can look at this again when DRYing and cleaning up the code and see if it makes sense to remove it, cool?